platform/security/sts/security-sts-upbstvalidator/src/main/java/org/codice/ddf/security/validator/uname/UPBSTValidator.java

```diff
@@ -198,25 +213,28 @@ public TokenValidatorResponse validateToken(TokenValidatorParameters tokenParame
         UsernameTokenType usernameTokenType = getUsernameTokenType(usernameToken);
         // Marshall the received JAXB object into a DOM Element
         Element usernameTokenElement = null;
-        try {
-            Set<Class<?>> classes = new HashSet<Class<?>>();
-            classes.add(ObjectFactory.class);
-            classes.add(
-                    org.apache.cxf.ws.security.sts.provider.model.wstrust14.ObjectFactory.class);
-
-            JAXBContextCache.CachedContextAndSchemas cache = JAXBContextCache
-                    .getCachedContextAndSchemas(classes, null, null, null, false);
-            JAXBContext jaxbContext = cache.getContext();
-
-            Marshaller marshaller = jaxbContext.createMarshaller();
-            Document doc = DOMUtils.createDocument();
-            Element rootElement = doc.createElement("root-element");
-            JAXBElement<UsernameTokenType> tokenType = new JAXBElement<UsernameTokenType>(
-                    QNameConstants.USERNAME_TOKEN, UsernameTokenType.class, usernameTokenType);
-            marshaller.marshal(tokenType, rootElement);
+
+        JAXBElement<UsernameTokenType> tokenType = new JAXBElement<>(QNameConstants.USERNAME_TOKEN,
+                UsernameTokenType.class,
+                usernameTokenType);
+
+        Document doc = DOMUtils.createDocument();
+        Element rootElement = doc.createElement("root-element");
+
+        List<String> ctxPath = ImmutableList.of(UsernameTokenType.class.getPackage()
+                .getName());
+
+        if (null != parser) {
+            ParserConfigurator configurator = parser.configureParser(ctxPath,
+                    UPBSTValidator.class.getClassLoader());
+            try {
+                parser.marshal(configurator, tokenType, rootElement);
+            } catch (ParserException ex) {
+                LOGGER.warn("", ex);
+                return response;
+            }
             usernameTokenElement = (Element) rootElement.getFirstChild();
-        } catch (JAXBException ex) {
-            LOGGER.warn("", ex);
+        } else {
             return response;
```

Comment:
It's not correct to return the response here; if the parser is null, that's an illegal state.